src/test/test-locale-util.c

```diff
@@ -50,9 +50,42 @@ static void test_locale_is_valid(void) {
         assert_se(!locale_is_valid("\x01gar\x02 bage\x03"));
 }
 
+static void test_get_keymaps(void) {
+        _cleanup_strv_free_ char **kmaps = NULL;
+        char **p;
+        int r;
+
+        r = get_keymaps(&kmaps);
+/*
+        assert_se(r >= 0);
+        assert_se(kmaps);
+
+        STRV_FOREACH(p, kmaps) {
+                puts(*p);
+                assert_se(keymap_is_valid(*p));
+        }
+*/
+}
+
+static void test_keymap_is_valid(void) {
+/*
+        assert_se(keymap_is_valid("uk"));
+        assert_se(keymap_is_valid("de-nodeadkeys"));
+        assert_se(keymap_is_valid("ANSI-dvorak"));
+        assert_se(keymap_is_valid("unicode"));
+*/
```

Comment:
i think it would make a ton of sense, to do these four lines only if get_keymaps() returned != -ENOENT. For that'd I'd just merge test_get_keymaps() and test_keymap_is_valid() into a single function, testing both APIs